### PR TITLE
[front] Add migration script for dust apps

### LINKED ISSUE #7243: [Vaults] Migrate apps savedConfig
Datasource, database and database_schema blocks use in their config references to datasources. This is currently stored under data_source_id and was previously the name of the datasource. From now on we store the datasource view id - we should migrate all existing apps with the sId of the view associated to the datasource.

### PR DESCRIPTION
**Pull request by tdraier**
fixes: https://github.com/dust-tt/dust/issues/7243

## Description

Migrate existing dust apps to use data source view sId in their config instead of datasource name

## Risk

break existing configs 

## Deploy Plan

run migration script

### CODE REVIEW

**Review by spolu — COMMENTED** [review]
Left 1 comment otherwise looks great

File: front/migrations/20240910_app_data_sources.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        DataSourceResource.fetchByNameOrId(auth, dataSource)
+      )
+    )
+  );
+
+  const dataSourceViews: Record<string, DataSourceViewResource> = (
+    await DataSourceViewResource.listForDataSourcesInVault(
+      auth,
+      dataSources,
+      globalVault
```
[the remark this answers is not in the document]
Comment (tdraier) [reply front/migrations/20240910_app_data_sources.ts]: Updated to iterate on available vaults, and list apps for each vault. Don't use globalVault anymore

**Review by spolu — APPROVED** [review]
I would save all previous specifications/config to a file or pipe it to the output as JSONL so that we have a way to revert in case this goes badly

Otherwise LGTM

**Comment by tdraier** [comment]
> I would save all previous specifications/config to a file or pipe it to the output as JSONL so that we have a way to revert in case this goes badly
> 
> Otherwise LGTM

Good point, I will add that ..

**Pull request merged**